### PR TITLE
Lib index

### PR DESCRIPTION
**Pull request by khanzf**
This reads the .git/index file (called the cache by GPL git).
There is a lot of string manipulation that could be cleaned up, but please verify that there are no overt problems.

### CODE REVIEW

File: lib/index.c [hunk 1]
```diff
@@ -30,120 +30,170 @@
...
-    This code is very confusing and is worse on the GPL git side.
-    This is the best I could do to make it readable.
-*/
-struct cache_tree *
-parse_treeentries(unsigned char *indexmap, int *offset)
+ * Description: Captures the cache tree data
+ * ToFree: Requires treeleaf->subtree to be freed
+ */
+static struct treeleaf *
+tree_entry(unsigned char *indexmap, long *offset, int extsize)
```
Comment (ibara) [inline lib/index.c]: `off_t`, not `long`

File: lib/index.c [hunk 2]
```diff
@@ -30,120 +30,170 @@
...
+	return (treeleaf);
 }
 
-void
-parse_indexentries(unsigned char *indexmap, int *offset, int entries)
+/*
+ * Must free: dircleaf
+ */
+static struct dircleaf *
+dirc_entry(unsigned char *indexmap, long *offset, int entries)
```
Comment (ibara) [inline lib/index.c]: Same thing here. `off_t`, not `long`.

**Review by ibara — CHANGES_REQUESTED** [review]

File: lib/index.c (hunk 1 above)
Comment (khanzf) [reply lib/index.c]: Thank you!

File: lib/index.c (hunk 2 above)
Comment (khanzf) [reply lib/index.c]: Again, thank you for the review.

**Pull request merged**